src/main/java/com/google/devtools/build/lib/analysis/ConfigurationTransitionDependency.java

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.devtools.build.lib.analysis;
+
+import com.google.auto.value.AutoValue;
+import com.google.devtools.build.lib.analysis.config.transitions.ConfigurationTransition;
+import com.google.devtools.build.lib.cmdline.Label;
+
+/** Implementation of a dependency with a given configuration transition. */
+@AutoValue
+public abstract class ConfigurationTransitionDependency {
```

Comment:
What about DependencyInfo or DependencyKey? I support, as part of distinguishing this, distinguishing this a clearly as possible in the naming too. As long as it ends in *Dependency" is still reads to me like it's an example of a `Dependency`.

I'd personally be happy with either of the above suggestions. But I also want to acknowledge DependencyInfo could get a bit confused with the existing DependencyKind  and DependencyKey implies the concept of Sky keys. 

But even if not perfect, let's give it a name that doesn't end in "Dependency".

Reply:
Changed to DependencyKey, let me know how this reads.